kernel/sched.c

```diff
@@ -1338,45 +1338,64 @@ struct k_thread *z_priq_rb_best(struct _priq_rb *pq)
 }
 
 #ifdef CONFIG_SCHED_MULTIQ
-# if (K_LOWEST_THREAD_PRIO - K_HIGHEST_THREAD_PRIO) > 31
-# error Too many priorities for multiqueue scheduler (max 32)
-# endif
+
+struct prio_info {
+	uint8_t offset_prio;
+	uint8_t idx;
+	uint8_t bit;
+};
+
+static inline struct prio_info get_prio_info(int8_t old_prio)
+{
+	struct prio_info ret;
+
+	ret.offset_prio = old_prio - K_HIGHEST_THREAD_PRIO;
+	ret.idx = ret.offset_prio & ~(NBITS - 1);
```

Comment:
I think the calculation (and subsequent use) of the `idx` field is incorrect. As it currently stands, it is only clearing the lowest (NBITS - 1) bits. To use it later as an index into `pq->bitmask[]`, a right shift (or division) is required; otherwise, as it currently stands, when it is used as an index, it may exceed the bounds of the `bitmask[]` array.

Reply:
Fixed. Thank for review.